[Chore] Liquibase 적용 및 ChangeSet 작성

isak-kang opened the pull request:
## 🛠️ Issue Number
### closes #40 

📌 **작업 내용 및 특이사항**
- DB 형상 관리를 위한 Liquibase 적용
- 기본 상점 ChangeSet 추가


📚 **참고사항**
- 내일(11.17) 상품, 리뷰 등으로 확장

isak-kang committed:
liquibase 의존성 추가

isak-kang committed:
gradle liquibase 플러그인 추가

isak-kang committed:
liquibase.properties 파일 .gitignore 설정

isak-kang committed:
default-db-management 프로필 추가

isak-kang committed:
create table, insert data changeset 작성 및 Entrypoint 마스터 파일 작성

isak-kang committed:
insert data의 updated_by,created_by data type수정

isak-kang committed:
chore: insert-data 컬럼 타입 수정 및 store_id 추가

isak-kang committed:
spotless 적용

willjsw commented on src/main/resources/application.yaml:
```diff
@@ -3,9 +3,9 @@ spring:
-      "prod": "prod, prod-datasource, prod-data-redis, prod-cache-redis, eureka, openfeign, actuator"
+      "local": "local, default-datasource, default-db-management, default-redis, eureka, openfeign, actuator"
+      "docker": "docker, default-datasource, default-db-management, default-redis, eureka, openfeign, actuator"
+      "prod": "prod, prod-datasource, default-db-management, prod-data-redis, prod-cache-redis, eureka, openfeign, actuator"
```
P1: prod 프로필 활성화 시에는 prod-db-management 프로필 적용해야 합니다!

willjsw commented on src/main/resources/db/changelog/changes/002-insert-default-store.yaml:
```diff
@@ -0,0 +1,48 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-data-2
```
P4: changeSet Id는 사실 어떤 방식으로 작성해도 고유성만 보장된다면 상관 없습니다!
다만 제가 처음 작성 시 의도했던 것은 첫번째 데이터 초기화라면 ` insert-data-1` 로 설정하고, "데이터 입력 행위의 1번째 변경이력" 이라는 의미를 부여하는 것이었긴 합니다😅
참고해주시면 감사하겠습니다!

willjsw reviewed (changes requested):
빠르게 작업해주셔서 감사합니다! 프로필 그룹 설정만 수정해주세요~

isak-kang committed:
리뷰사항 적용

isak-kang committed:
카테고리 데이터셋 작성 초안

growingpsy committed:
Merge branch 'develop' into chore/#40-chore-liquibase

isak-kang committed:
Merge branch 'develop' into chore/#40-chore-liquibase

growingpsy committed:
안쓰이는 redisconfig 제거

isak-kang committed:
Merge branch 'chore/#40-chore-liquibase' of https://github.com/Profect-4th-IRUM/product-service into chore/#40-chore-liquibase

isak-kang committed:
Liquibase 위치 수정

growingpsy committed:
fix #40: RedisRepositorySliceTest에 @ActiveProfiles("test") 어노테이션 추가

growingpsy committed:
chore #40: securitytestconfig 삭제

growingpsy committed:
refactor #40: testconfig 재작성

isak-kang committed:
store 데이터셋 수정

isak-kang committed:
카테고리 changes 데이터셋 추가

isak-kang committed:
product에서 필요한 테이블 생성

isak-kang committed:
master 추가

isak-kang committed:
Merge branch 'chore/#40-chore-liquibase' of https://github.com/Profect-4th-IRUM/product-service into chore/#40-chore-liquibase

isak-kang committed:
char -> varchar로 변경 데이터타입(telemarketing_registration_number,business_registration_number)

isak-kang committed:
spotless적용

isak-kang committed:
Revert "spotless적용"

This reverts commit 671688f3e54143a318665dd987ef8a92ba7e3315.

growingpsy committed:
chore #40: PROFILE_ACTIVE:local 적용

growingpsy committed:
chore #40: ProductCursorResponse 경로 수정

growingpsy committed:
refactor #40: 컨트롤러 테스트 전면 수정

growingpsy committed:
style #40: spotlessApply

growingpsy committed:
chore #40: liquibase false

growingpsy committed:
chore #40: ProductInternalIntegrationTest.java 주석처리

growingpsy committed:
chore #40: ProductInternalIntegrationTest 동시성/롤백 통합 테스트 안정화 및 데이터 충돌 해결

- Store 테스트 데이터 생성 시 고정 memberId 및 고정 사업자번호로 인해
  유니크 제약(p_store_member_key / business_registration_number) 충돌 발생 → 수정
- 테스트용 Store 생성 시 random memberId 및 random 사업자번호 사용하도록 변경
- 테스트 간 데이터 누적에 따른 중복 INSERT 문제 해결
- ProductOptionValue 조회/삭제 시 OptimisticLock 예외 발생 문제 해결
  (테스트 종료 후 삭제 로직 제거 및 조회 기반 검증 방식으로 전환)
- 순차 주문 / 동시 주문 / 순차 롤백 / 동시 롤백 테스트 전체 안정화
- 최종 재고 검증 로직 강화 (DB에서 OptionValue 조회하여 검증)
- 동시성 테스트 스레드 풀 및 CountDownLatch 적용 안정화
- 예외 종류별 카운트 로직 정리 (재고 부족 / 재시도 초과 / 기타 예외 분리)

결과적으로 실제 운영 환경에서 발생 가능한 재고 차감·복구 동시성 시나리오를
SpringBootTest + 실제 DB 기반으로 완전하게 재현하도록 개선.

isak-kang committed:
카테고리, 상품, 상품옵션값,그룹, 이미지, 배송비정책, 할인,리뷰, 리뷰이미지 데이터셋

isak-kang committed:
Merge branch 'chore/#40-chore-liquibase' of https://github.com/Profect-4th-IRUM/product-service into chore/#40-chore-liquibase

isak-kang committed:
spotless 적용

willjsw reviewed (approved):
LGTM
고생하셨습니다:)
